pkg/stanza/fileconsumer/file.go

```diff
@@ -106,31 +105,24 @@ func (m *Manager) startPoller(ctx context.Context) {
 
 // poll checks all the watched paths for new entries
 func (m *Manager) poll(ctx context.Context) {
-	var matches []string
-	if len(m.queuedMatches) > m.maxBatchFiles {
-		matches, m.queuedMatches = m.queuedMatches[:m.maxBatchFiles], m.queuedMatches[m.maxBatchFiles:]
-	} else {
-		if len(m.queuedMatches) > 0 {
-			matches, m.queuedMatches = m.queuedMatches, make([]string, 0)
-		} else {
-			// Increment the generation on all known readers
-			// This is done here because the next generation is about to start
-			for i := 0; i < len(m.knownFiles); i++ {
-				m.knownFiles[i].generation++
-			}
-
-			// Get the list of paths on disk
-			matches = m.finder.FindFiles()
-			if len(matches) > m.maxBatchFiles {
-				matches, m.queuedMatches = matches[:m.maxBatchFiles], matches[m.maxBatchFiles:]
-			}
-		}
+	// Increment the generation on all known readers
+	// This is done here because the next generation is about to start
+	for i := 0; i < len(m.knownFiles); i++ {
+		m.knownFiles[i].generation++
 	}
 
-	readers := m.makeReaders(matches)
+	// Get the list of paths on disk
+	matches := m.finder.FindFiles()
+	for len(matches) > m.maxBatchFiles {
+		m.consume(ctx, matches[:m.maxBatchFiles])
```

Comment:
If I am understanding this correctly, the `poll` function will now consume (in batches) _all_ know files immediately, where as before it only did 1 batch?

Reply:
Correct. It will still use batching to respect the `max_concurrent_files` setting, but it will process all batches within a single poll. Prior to this change it would process one batch per poll.